dali/base/sysinfologger.cpp

```diff
@@ -23,42 +23,45 @@
 #define SYS_INFO_VERSION "1.0"
 
 #define SYS_INFO_ROOT  "/SysLogs"
-#define ATTR_NEXTID    "@nextId"
+#define ATTR_NEXTSEQN  "@nextSeqNum"
 #define ATTR_VERSION   "@version"
 #define MSG_NODE       "msg"
-#define ATTR_ID        "@id"
+#define ATTR_SEQNUM    "@seqNum"
 #define ATTR_TIMESTAMP "@ts"
 #define ATTR_AUDIENCE  "@audience"
 #define ATTR_CLASS     "@class"
 #define ATTR_CODE      "@code"
 #define ATTR_HIDDEN    "@hidden"
 #define ATTR_SOURCE    "@source"
 
+#define NON_SYSLOGMSG_MASK 0x8000000000
+
 static void extractDate(timestamp_type ts, unsigned & year, unsigned & month, unsigned & day)
 {
     CDateTime timeStamp;
     timeStamp.setTimeStamp(ts);
     timeStamp.getDate(year, month, day);
 }
 
-static unsigned __int64 makeMessageId(unsigned year, unsigned month, unsigned day, unsigned id)
+unsigned __int64 makeMessageId(unsigned year, unsigned month, unsigned day, unsigned seqN, bool nonSysInfoLogMsg)
 {
-    return id<<21 | year<<9 | month<<5 | day;
+    return (nonSysInfoLogMsg?NON_SYSLOGMSG_MASK:0) | seqN<<21 | year<<9 | month<<5 | day;
```

Comment:
not new/minor: bits 9-20 inclusive reserved for year (12 bits), so max year is 4095?
Not that it's ever going to an issue :) , but potentially confusing to reader, should cope with any 4 digit year, and needs 14 bits for that. so would have to be seqN<<23 I think.

Arguably might be better to more explicitly acknowledge that the sequence number is 32bit only and shift it to the top 4 bytes, and use the spare bits in the lower unsigned space for everything else:
```
    return seqN<<32 | (nonSysInfoLogMsg?nonSysInfoLogMsgMask:0) | year<<9 | month<<5 | day;
```
where nonSysInfoLogMsgMask uses (e.g.) 23th bit

Probably worth adding constexpr for day/month/year masks, with comments to bit range for clarify.

Reply:
1) I'm not sure that max year 4095 is confusing.  9999 is another arbitrary upper limit just like 4095.  (It is just as valid to have year 10000 year 9999!)
2) If supporting all 4 digit years is important, we need the nonSysInfoLogMsgMask at 24 bit (not bit 23)
3) If we have the nonSysInfoLogMsgMask as the 32 bit, it would allow other bits between 25-31 to be be available for other uses.

@jakesmith

Reply:
you're right 9999 is just as arbitrary - I think as a long as it's clear. Add comments that using 12 bits and what upper limit is and it'll be fine (I think supporting 0-9999 would be conventional though).

>If supporting all 4 digit years is important, we need the nonSysInfoLogMsgMask at 24 bit (not bit 23)

yes 24th bit:

0-4 = day
5-8 = month
9-22 = if 14 bits for year [ currently 9-20 (12 bits) ]
23-31 = for other meta, e.g. 1 bit to signify static messages.
32-63 = upper 32 bits for sequence number.

would be worth adding this as a comment block (near the mask definition) and/or adding masks for other for others together with the comments.